data/users.js

```diff
@@ -1,20 +1,79 @@
-import {users} from '../config/mongoCollections.js';
-import {ObjectId} from 'mongodb';
-import validation from '../validation.js';
-
-let exportedMethods = {
-    async getAllUsers() {
-      const userCollection = await users();
-      const userList = await userCollection.find({}).toArray();
-      return userList;
-    },
-    async getUserById(id) {
-      id = validation.checkValidId(id,"user_id");
-      const userCollection = await users();
-      const user = await userCollection.findOne({_id: ObjectId(id)});
-      if (!user) throw 'Error: User not found';
-      return user;
-    }
-};
-
-export default exportedMethods;
\ No newline at end of file
+import { users } from '../config/mongoCollections.js';
+import { ObjectId } from 'mongodb';
+import validation from '../validation.js';
+import { validationsForSignUp, validationsForLogin } from '../public/js/validators/user.js';
+import * as bcrypt from 'bcrypt';
+
+/** 
+ * A constant global variable which will be called only once
+ * to get the userCollection and from there on you can just reuse the 
+ * variable which will save a lot of time in every call since we cut
+ * down one promise in our flow.
+ */
+const userCollection = await users();
+
+const getAllUsers = async () => {
+	const userCollection = await users();
+	const userList = await userCollection.find({}).toArray();
+	return userList;
+};
+
+const getUserById = async (id) => {
+	id = validation.checkValidId(id, "user_id");
+	const userCollection = await users();
+	const user = await userCollection.findOne({ _id: ObjectId(id) });
+	if (!user) throw 'Error: User not found';
+	return user;
+}
+
+/**
+ * This function will hash the password and create a new user doc in db
+ * 
+ * @param {{ 
+ * 	firstName: String, 
+ * 	lastName: String, 
+ * 	email: String, 
+ * 	age: Number, 
+ *  username: String, 
+ *  password: String 
+ * }} user 
+ * 
```

Comment:
This is called as jsdoc and it will be very helpful in documenting code, hovering on the function in vscode will give a really good description of what that function is as if it is a library function

![image](https://user-images.githubusercontent.com/40670894/231597501-0bef1fd0-355a-4d8d-a7fb-6268dd173c03.png)